src/test/java/org/apache/commons/compress/archivers/tar/TarArchiveEntryTest.java

```diff
@@ -306,6 +308,164 @@ public void getOrderedSparseHeadersRejectsStructsPointingBeyondOutputEntry() thr
         te.getOrderedSparseHeaders();
     }
 
+    @Test
+    public void shouldParseTimePaxHeadersAndNotCountAsExtraPaxHeaders() {
+        TarArchiveEntry entry = createEntryForTimeTests();
+        assertEquals("extra header count", 0, entry.getExtraPaxHeaders().size());
+        assertNull("size", entry.getExtraPaxHeader("size"));
+        assertNull("mtime", entry.getExtraPaxHeader("mtime"));
+        assertNull("atime", entry.getExtraPaxHeader("atime"));
+        assertNull("ctime", entry.getExtraPaxHeader("ctime"));
+        assertNull("birthtime", entry.getExtraPaxHeader("LIBARCHIVE.creationtime"));
+        assertEquals("size", entry.getSize(), 1);
+        assertEquals("mtime", FileTime.from(Instant.parse("2022-03-14T01:25:03.599853900Z")), entry.getLastModifiedTime());
+        assertEquals("atime", FileTime.from(Instant.parse("2022-03-14T01:31:00.706927200Z")), entry.getLastAccessTime());
+        assertEquals("ctime", FileTime.from(Instant.parse("2022-03-14T01:28:59.700505300Z")), entry.getStatusChangeTime());
+        assertEquals("birthtime", FileTime.from(Instant.parse("2022-03-14T01:29:00.723509000Z")), entry.getCreationTime());
+    }
+
+    @Test
+    public void shouldNotWriteTimePaxHeadersByDefault() throws IOException {
+        TarArchiveEntry entry = createEntryForTimeTests();
+
+        final ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        final TarArchiveOutputStream tos = new TarArchiveOutputStream(bos);
+        tos.putArchiveEntry(entry);
+        tos.write('W');
+        tos.closeArchiveEntry();
+        tos.close();
+
+        final TarArchiveInputStream tis = new TarArchiveInputStream(new ByteArrayInputStream(bos.toByteArray()));
+        entry = tis.getNextTarEntry();
+        assertNotNull("couldn't get entry", entry);
+
+        assertEquals("extra header count", 0, entry.getExtraPaxHeaders().size());
+        assertNull("mtime", entry.getExtraPaxHeader("mtime"));
+        assertNull("atime", entry.getExtraPaxHeader("atime"));
+        assertNull("ctime", entry.getExtraPaxHeader("ctime"));
+        assertNull("birthtime", entry.getExtraPaxHeader("LIBARCHIVE.creationtime"));
+        assertEquals("mtime", FileTime.from(Instant.parse("2022-03-14T01:25:03Z")), entry.getLastModifiedTime());
+        assertNull("atime", entry.getLastAccessTime());
+        assertNull("ctime", entry.getStatusChangeTime());
+        assertNull("birthtime", entry.getCreationTime());
+
+        assertEquals('W', tis.read());
+        assertTrue("should be at end of entry", tis.read() < 0);
+
+        assertNull("should be at end of file", tis.getNextTarEntry());
+        tis.close();
+    }
+
+    @Test
+    public void shouldWriteTimesForStarMode() throws IOException {
+        TarArchiveEntry entry = createEntryForTimeTests();
+
+        final ByteArrayOutputStream bos = new ByteArrayOutputStream();
```

Comment:
Use try-with-resources. I know this is will not cause a real resource leak on a thrown exception this avoids compiler warnings and it is a best practice we want to exemplify.

Reply:
Done